mne_bids/read.py

```diff
@@ -619,22 +627,28 @@ def get_head_mri_trans(bids_path, extra_params=None):
                            'instantiate using mne_bids.BIDSPath().')
 
     # check root available
-    bids_path = bids_path.copy()
-    bids_root = bids_path.root
-    if bids_root is None:
+    meg_bids_path = bids_path.copy()
+    del bids_path
+    if meg_bids_path.root is None:
         raise ValueError('The root of the "bids_path" must be set. '
                          'Please use `bids_path.update(root="<root>")` '
                          'to set the root of the BIDS folder to read.')
+
     # only get this for MEG data
-    bids_path.update(datatype='meg')
+    meg_bids_path.update(datatype='meg', suffix='meg')
 
     # Get the sidecar file for MRI landmarks
-    bids_fname = bids_path.update(suffix='meg', root=bids_root)
-    t1w_json_path = _find_matching_sidecar(bids_fname, suffix='T1w',
-                                           extension='.json')
+    if t1_bids_path is None:
+        t1w_json_path = _find_matching_sidecar(meg_bids_path, suffix='T1w',
+                                               extension='.json')
+    else:
+        t1_bids_path = t1_bids_path.copy().update(suffix='T1w',
+                                                  datatype='anat')
+        t1w_json_path = _find_matching_sidecar(t1_bids_path, suffix='T1w',
+                                               extension='.json')
 
     # Get MRI landmarks from the JSON sidecar
-    with open(t1w_json_path, 'r', encoding='utf-8-sig') as f:
+    with open(t1w_json_path, 'r', encoding='utf-8') as f:
```

Comment:
why did you change this?

Reply:
Good catch, forgot to mention this!

So it SHOULD have been `utf-8` all along; the UTF BOM character which is handled by `utf-8-sig` is invalid in JSON. Meaning that we shouldn't even try to read it because … well, it shouldn't be present!

One could argue that this change could break backward-compat for some users; however their datasets wouldn't be BIDS-compatible anyway if their JSON files contain a BOM (as, like I said, the BOM is invalid in JSON)

In all other places of MNE-BIDS, we read & write JSON using `utf-8` these days, not `utf-8-sig`; I suppose this line here was just overlooked in #625